itou/common_apps/openid_connect/models.py

```diff
@@ -43,15 +43,18 @@ class OIDConnectUserData:
     def create_or_update_user(self):
         """
         Create or update a user managed by another identity provider.
-         - If there is already a user with this username (user_info_dict["sub"]), we update and return it.
+         - If there is already a user with this username (user_info_dict["sub"])
+           and from the same identity provider, we update and return it.
          - If there is already a user with the email, we return this user.
          - otherwise, we create a new user based on the data we received.
         """
         user_data_dict = dataclasses.asdict(self)
         user_data_dict = {key: value for key, value in user_data_dict.items() if value}
         try:
+            # If the same username exists with another identity_provider, it may not be the same user
+            # so don't update and return if !
```

Comment:
> and return if !

? C'est du grand pinaillage mais j'ai l'impression qu'il manque un mot dans ce commentaire.

Reply:
Oups, c'est une coquille : `if` -> `it`